python/xir/parser.py

```diff
@@ -183,26 +189,42 @@ def statement(self, args):
     def gatestmt(self, args):
         """Gate statements that are defined directly in the circuit or inside
         a gate declaration."""
+
+        adjoint = False
+        ctrl_wires = tuple()
+
+        while args[0] in ("adjoint", "ctrl"):
+            a = args.pop(0)
+            if a == "adjoint":
+                adjoint = not adjoint
+            elif a == "ctrl":
+                ctrl_wires += args.pop(0)[1]
+
         name = args[0]
         if isinstance(args[1], list):
-            params = simplify_math(args[1])
-            wires = args[2]
+            params = [simplify_math(a) for a in args[1]]
```

Comment:
No need to change anything but I would just like to point out this is a good use case for `map`!
```suggestion
            params = list(map(simplify_math, args[1]))
```